src/zc/buildout/easy_install.py

```diff
@@ -1658,7 +1666,117 @@ def call_easy_install(spec, dest):
-    return glob.glob(os.path.join(dest, '*'))
+
+    split_entries = [os.path.splitext(entry) for entry in os.listdir(dest)]
+    try:
+        distinfo_dir = [
+            base + ext for base, ext in split_entries if ext == ".dist-info"
+        ][0]
+    except IndexError:
+        logger.error(
+            "No .dist-info directory after installing %s",
+            spec)
+        raise
+
+    return make_egg_after_pip_install(dest, distinfo_dir)
+
+
+def make_egg_after_pip_install(dest, distinfo_dir):
+    """build properly named egg directory"""
+
+    # `pip install` does not build the namespace aware __init__.py files
+    # but they are needed in egg directories.
+    # Add them before moving files setup by pip
+    ns_file = os.path.join(dest, distinfo_dir, 'namespace_packages.txt')
+    if os.path.exists(ns_file):
+        with open(ns_file) as f:
+            namespace_packages = [
+                line.strip().replace('.', os.path.sep)
+                for line in f.readlines()
+            ]
+
+        for namespace_package in namespace_packages:
+            init_py_file = os.path.join(dest, namespace_package, '__init__.py')
+            with open(init_py_file, 'w') as f:
+                f.write(
+                    "__import__('pkg_resources').declare_namespace(__name__)"
+                )
+
+    # Remove `bin` directory if needed
+    # as there is no way to avoid script installation
+    # when running `pip install`
+    ep_file = os.path.join(dest, distinfo_dir, 'entry_points.txt')
+    if os.path.exists(ep_file):
+        with open(ep_file) as f:
+            if "console_scripts" in f.read():
+                bin_dir = os.path.join(dest, 'bin')
+                if os.path.exists(bin_dir):
+                    shutil.rmtree(bin_dir)
+
+    # Make properly named new egg dir
+    distro = [d for d in pkg_resources.find_distributions(dest)][0]
+    egg_name = distro.egg_name() + '.egg'
+    egg_dir = os.path.join(dest, egg_name)
+    os.mkdir(egg_dir)
+
+    # Move ".dist-info" dir into new egg dir
+    os.rename(
+        os.path.join(dest, distinfo_dir),
+        os.path.join(egg_dir, distinfo_dir)
+    )
+
+    with open(os.path.join(egg_dir, distinfo_dir, 'top_level.txt')) as f:
+        top_levels = filter(
+            (lambda x: len(x) != 0),
+            [line.strip() for line in f.readlines()]
+            )
+
+    # Move all top_level modules or packages
+    for top_level in top_levels:
+        # as package
+        top_level_dir = os.path.join(dest, top_level)
+        if os.path.exists(top_level_dir):
+            shutil.move(top_level_dir, egg_dir)
+            continue
+        # as module
+        top_level_py = top_level_dir + '.py'
+        if os.path.exists(top_level_py):
+            shutil.move(top_level_py, egg_dir)
+            top_level_pyc = top_level_dir + '.pyc'
+            if os.path.exists(top_level_pyc):
+                shutil.move(top_level_pyc, egg_dir)
+            continue
```

Comment:
What about `__pycache__` subdirectories?

Reply:
On second thought, it's fine to not copy those.

Now I'm thinking about filenames embedded in pyc files.  Do we want to copy pyc files?  Maybe, if there happens to be a binary-only module where there are pyc files without matching py files (a use case heavily frowned upon, but supported, even on Python 3).